### DISCUSSION TITLE
doris导出查询结果集到 minio 报错

### DISCUSSION THREAD

**Discussion by ly137062117** [apache/doris#8359] [A - General / Q&A] [answered]
doris 官方文档上支持导出到兼容 S3协议的存储系统，我自建了一个 minio，并且开启了 S3兼容模式，然后通过 Doris 导出查询结果集报错，『RpcException, msg: send fragment timeout. backend id: xxx, host: xxx』

查看 be 的日志，错误信息：
[AWSClient] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
[AWSClient] Request failed, now waiting 3200 ms before attempting again.
[CurlHttpClient] Curl returned error code 6 - Couldn't resolve host name

请求的sql：
select * from xxx 
INTO OUTFILE "s3://test/export_file_"
FORMAT AS CSV
PROPERTIES
(
    "AWS_ENDPOINT" = "http://${IP_ADDRESS}:${PORT}",
    "AWS_ACCESS_KEY"="xxx",
    "AWS_SECRET_KEY" = "xxx"
);

AWS_ENDPOINT是只能配置域名吗？修改了本地 hosts 文件也不行，还是目前 doris 导出查询结果集，只支持导出到公有云供应商提供的存储系统？类似自建的 minio 这种，即便兼容了 S3，也不行

**Comment by hf200012** [comment]
From the information it looks like a network problem, whether your doris cluster and minio's network is open, whether the firewall is open

**Reply by ly137062117** [reply]
Yes, I'm sure minio's network is open and accessible.

using 
create repository s_repo with s3 on location "s3://xxx/xxx" properties ("AWS_ENDPOINT" = "http://xxx:8902", "AWS_ACCESS_KEY"="xxx", "AWS_SECRET_KEY" = "xxx","AWS_REGION" = "xxx", "use_path_style"="true")

I could connect to my minio and create a repository successfully, but using select into outfile with the same parameter failed

I guess there may be some reason about the parameter "use_path_style" ?

**Comment by ly137062117** [comment, marked as answer]
已解决，minio 需要通过 export MINIO_DOMAIN=mydomain.com 开启 virtual host，然后在 doris be 所在的机器上配置 /etc/hosts，添加 mydomain.com 以及 ${BUCKET}.mydomain.com 到 minio server 的 ip 映射

**Reply by hf200012** [reply]
> 已解决，minio 需要通过 export MINIO_DOMAIN=mydomain.com 开启 virtual host，然后在 doris be 所在的机器上配置 /etc/hosts，添加 mydomain.com 以及 ${BUCKET}.mydomain.com 到 minio server 的 ip 映射

Is it convenient to add WeChat, my WeChat: 35926237

**Reply by leixw0102** [reply]
> 已解决，minio 需要通过 export MINIO_DOMAIN=mydomain.com 开启 virtual host，然后在 doris be 所在的机器上配置 /etc/hosts，添加 mydomain.com 以及 ${BUCKET}.mydomain.com 到 minio server 的 ip 映射

hi,我也遇到同样的问题 ，doris导入s3 。可以详细说一下么。PROPERTIES (
    "AWS_ENDPOINT" = "http://ip:port",
    "AWS_ACCESS_KEY" = "",
    "AWS_SECRET_KEY" = "",
    "AWS_REGION"=""
  );

这块需要在doiris be上添加 exaple.com   ----> ip:port 就可以么？
